test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,552 @@ describe("BaseAxis", () => {
+              TestMethods.numAttr(firstAnnotationRect, "height"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+
+          it("moves rectangles that would overlap to different rows upwards (orientation: top)", () => {
+            let axis = new Plottable.Axis(scale, "top");
+            axis.annotatedTicks([50, 51]);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "y") - TestMethods.numAttr(secondAnnotationRect, "y"),
+              TestMethods.numAttr(firstAnnotationRect, "height"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+
+          it("moves rectangles that would overlap to different rows leftwards (orientation: left)", () => {
+            let axis = new Plottable.Axis(scale, "left");
+            axis.annotatedTicks([50, 51]);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "x") - TestMethods.numAttr(secondAnnotationRect, "x"),
+              TestMethods.numAttr(firstAnnotationRect, "width"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+
+          it("moves rectangles that would overlap to different rows rightwards (orientation: right)", () => {
+            let axis = new Plottable.Axis(scale, "right");
+            axis.annotatedTicks([50, 51]);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(secondAnnotationRect, "x") - TestMethods.numAttr(firstAnnotationRect, "x"),
+              TestMethods.numAttr(firstAnnotationRect, "width"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+        });
+
+      });
+
+      it("hides annotations if rectangles are outside the margin area", () => {
+        let mockScale = new Plottable.Scales.Linear;
+        let annotatedTicks = [50, 51, 150];
+        mockScale.domain(annotatedTicks);
```

Comment:
Oddly enough, this works, but it shouldn't. This does set the domain to `[50, 51, 150]`, but that means that 50 maps to 0 and 51 maps to the `<svg>` width (300 in this case). I'd explicitly set the domain to `[0, SVG_WIDTH]`.